drivers/video/video_stm32_venc.c

```diff
@@ -0,0 +1,1026 @@
+}
+
+EWLHwConfig_t EWLReadAsicConfig(void)
+{
+	EWLHwConfig_t cfg_info;
+	u32 cfgval;
+
+	LOG_DBG("%s>\n", __func__);
+
+	/* read first part of the configuration stored in register 63 */
+	cfgval = LL_VENC_ReadRegister(63UL);
+
+	cfg_info.maxEncodedWidth = cfgval & ((1U << 12U) - 1U);
+	cfg_info.h264Enabled = (cfgval >> 27U) & 1U;
+	cfg_info.vp8Enabled = (cfgval >> 26U) & 1U;
+	cfg_info.jpegEnabled = (cfgval >> 25U) & 1U;
+	cfg_info.vsEnabled = (cfgval >> 24U) & 1U;
+	cfg_info.rgbEnabled = (cfgval >> 28U) & 1U;
+	cfg_info.searchAreaSmall = (cfgval >> 29U) & 1U;
+	cfg_info.scalingEnabled = (cfgval >> 30U) & 1U;
+
+	cfg_info.busType = (cfgval >> 20U) & 15U;
+	cfg_info.synthesisLanguage = (cfgval >> 16U) & 15U;
+	cfg_info.busWidth = (cfgval >> 12U) & 15U;
+
+	/* read second part of the configuration stored in register 296 */
+	cfgval = LL_VENC_ReadRegister(296UL);
+
+	cfg_info.addr64Support = (cfgval >> 31U) & 1U;
+	cfg_info.dnfSupport = (cfgval >> 30U) & 1U;
+	cfg_info.rfcSupport = (cfgval >> 28U) & 3U;
+	cfg_info.enhanceSupport = (cfgval >> 27U) & 1U;
+	cfg_info.instantSupport = (cfgval >> 26U) & 1U;
+	cfg_info.svctSupport = (cfgval >> 25U) & 1U;
+	cfg_info.inAxiIdSupport = (cfgval >> 24U) & 1U;
+	cfg_info.inLoopbackSupport = (cfgval >> 23U) & 1U;
+	cfg_info.irqEnhanceSupport = (cfgval >> 22U) & 1U;
+
+	LOG_DBG("EWLReadAsicConfig:\n");
+	LOG_DBG("    maxEncodedWidth   = %d\n", cfg_info.maxEncodedWidth);
+	LOG_DBG("    h264Enabled       = %d\n", cfg_info.h264Enabled);
+	LOG_DBG(         "    jpegEnabled       = %d\n", cfg_info.jpegEnabled);
+	LOG_DBG(         "    vp8Enabled        = %d\n", cfg_info.vp8Enabled);
+	LOG_DBG(         "    vsEnabled         = %d\n", cfg_info.vsEnabled);
+	LOG_DBG(         "    rgbEnabled        = %d\n", cfg_info.rgbEnabled);
+	LOG_DBG(         "    searchAreaSmall   = %d\n", cfg_info.searchAreaSmall);
+	LOG_DBG(         "    scalingEnabled    = %d\n", cfg_info.scalingEnabled);
+	LOG_DBG(         "    address64bits     = %d\n", cfg_info.addr64Support);
+	LOG_DBG(         "    denoiseEnabled    = %d\n", cfg_info.dnfSupport);
+	LOG_DBG(         "    rfcEnabled        = %d\n", cfg_info.rfcSupport);
+	LOG_DBG(         "    instanctEnabled   = %d\n", cfg_info.instantSupport);
+	LOG_DBG(         "    busType           = %d\n", cfg_info.busType);
+	LOG_DBG(         "    synthesisLanguage = %d\n", cfg_info.synthesisLanguage);
+	LOG_DBG(         "    busWidth          = %d\n", cfg_info.busWidth * 32);
+
+	LOG_DBG("%s<\n", __func__);
+	return cfg_info;
+}
+
+const void *EWLInit(EWLInitParam_t *param)
+{
+	LOG_DBG("%s>\n", __func__);
+
+	/* Check for NULL pointer */
+	__ASSERT_NO_MSG(param != NULL);
+	/* only H264 (0) and JPEG (1) are supported */
+	__ASSERT_NO_MSG(param->clientType <= 1U);
+
+	/* sync */
+	k_sem_init(&ewl_instance.complete, 0, 1);
+	k_sem_reset(&ewl_instance.complete);
+
+	/* set client type */
+	ewl_instance.clientType = param->clientType;
+
+	mem_attr_heap_pool_init();
+
+	ewl_instance.irq_cnt = 0;
```

Comment:
Nit: IMO, avoid mixing Zephyr and HW code
```suggestion
	/* set client type */
	ewl_instance.clientType = param->clientType;

	ewl_instance.irq_cnt = 0;

	mem_attr_heap_pool_init();
```
(I would even drop newline between the two top lines, but I kept it for the comment to be more scoped)